python/pyspark/ml/classification.py

```diff
@@ -1560,8 +1581,9 @@ def trainSingleClass(index):
                             (classifier.predictionCol, predictionCol)])
             return classifier.fit(trainingDataset, paramMap)
 
-        # TODO: Parallel training for all classes.
-        models = [trainSingleClass(i) for i in range(numClasses)]
+        pool = ThreadPool(processes=min(self.getParallelism(), numClasses))
```

Comment:
We limit the pool here, but not in Scala. Is there a real benefit to limiting it?